src/fmu/dataio/_utils.py

```diff
@@ -463,3 +463,35 @@ def glue_metadata_preprocessed(
     meta["tracklog"].extend(newmeta["tracklog"])
 
     return meta
+
+
+def get_geometry_ref(geometrypath: str | None, obj: Any) -> dict[str, str]:
```

Comment:
Would it be reasonable for this to return an instance of `Geometry` Pydantic object instead?